### ISSUE TITLE
Code Review: The functions 'hash5Tuple' and 'hash2Tuple' implemented incorrectly in PacketUtils.cpp.

### ISSUE THREAD

**Issue by prudens** [seladb/PcapPlusPlus#1447]
### Bug description

**Describe the bug**
I noticed that in the 'hash5Tuple' function, when the code logic reaches the branch where 'portDst == portSrc', it compares IPv6 and IPv4 addresses directly by converting them to 'uint64_t'. The correct approach seems to be: 'memcmp(ipv6Layer->getIPv6Header()->ipDst, ipv6Layer->getIPv6Header()->ipSrc, 16) < 0'. Additionally, it should check for the parameter '!directionUnique' at the beginning to ensure the functionality is effective when 'directionUnique' is 'true'.

**Code example to reproduce**
```cpp
in PacketUtil.cpp:202 
        if (portSrc == portDst && (uint64_t)ipv6Layer->getIPv6Header()->ipDst < (uint64_t)ipv6Layer->getIPv6Header()->ipSrc)
            srcPosition = 1;
```

**Expected behavior**
```cpp
        if (!directionUnique && portSrc == portDst && memcmp(ipv6Layer->getIPv6Header()->ipDst, ipv6Layer->getIPv6Header()->ipSrc, 16) < 0)
            srcPosition = 1;
```

### PcapPlusPlus versions tested on

PcapPlusPlus master branch

### Operating systems tested on

Linux

### Compiler version

Clang 15

### Packet capture backend (if applicable)

libpcap

**Comment by seladb** [comment]
@prudens good catch! This is indeed a bug, thank you for reporting it 🙏 
Would you consider opening a PR with a fix?

**Comment by prudens** [comment]
@seladb  I am very willing to submit a PR, but according to the requirements mentioned in the guidelines, my development environment is not yet properly configured to perform the necessary code checks and tests before submitting the PR.

**Comment by tigercosmos** [comment]
@prudens  what's the difficulty for you for now?

**Comment by seladb** [comment]
> ut according to the requirements mentioned in the guidelines, my development environment is not yet pr

@prudens I'm curious what is missing in your environment?
In the following links you can find instructions on how to set up the environment based on your system:
- Linux: https://pcapplusplus.github.io/docs/next/install/linux
- macOS: https://pcapplusplus.github.io/docs/next/install/macos
- Windows with Visual Studio: https://pcapplusplus.github.io/docs/next/install/vs
- Windows with MinGW: https://pcapplusplus.github.io/docs/next/install/mingw

**Comment by tigercosmos** [comment]
close due to no activity.

**Comment by egecetin** [comment]
Is this bug fixed? If not fixed, it should remain open

**Comment by prudens** [comment]
I will open a PR to fix it this week.

**Comment by seladb** [comment]
Fixed in PR #1495
